packages/docusaurus-theme-common/src/utils/useTOCHighlight.ts

```diff
@@ -144,6 +144,7 @@ function useTOCHighlight(config: TOCHighlightConfig | undefined): void {
         }
         link.classList.add(linkActiveClassName);
         lastActiveLinkRef.current = link;
+        link.scrollIntoView({block: 'nearest'});
```

Comment:
This breaks pages using `scroll-behavior: smooth` as both JS and CSS are trying to smoothly scroll the page